### ISSUE TITLE
[Question] Schema.prisma file

### ISSUE THREAD

**Issue by Jaspaul** [prisma/migrate#160]
I'm confused about the management of the `schema.prisma` file used with migrate.

I noticed that the `schema.prisma` file is copied from `prisma/schema.prisma` to each of the migrations in `prisma/migrations/...`. What concerns me about this is that each copy of the `schema.prisma` file contains a copy of the `datasource db {}` and `generator photon {}` blocks which may be changing overtime. For instance if I change my database name, or password, or the location where photon is generated these files will end up being inconsistent. 

Is this intentional? The behaviour I was expecting was that the `prisma/schema.prisma` file would contain these blocks and they would not be copied over to each of the migrations.

As I've been working with Prisma2 locally I've run into confusion around where my files are being generated and discovering that the following block was different in various `schema.prisma` files.

```prisma
generator photon {
  provider = "photonjs"
  output   = "src/generated"
}
```

I also ran into issues around this early on, during experimentation, when I hardcoded the datasource block and eventually wanted to change it to the following:

```prisma
datasource db {
  provider = env("SOME_COOL_PROVIDER")
  url      = env("SOME_COOL_URL")
}
```

Is there documentation available around how these values are expected to be used? Should the provider and url be configurable, since the queries that are run seem to be included in the README.md files in the migrations?

**Comment by janpio** [comment]
Interesting problem, that we probably were not aware of yet. Thanks for taking the time to create an issue.

I will put this in front of some people that might give some more insight and thoughts about this.

**Comment by mavilein** [comment]
Hey @Jaspaul ,

thanks for reaching out to us! 🙏 
You are right that the `schema.prisma` is copied into each migration folder that gets created. However the _contents of this file are not important_ as we are not using them for anything. We just copy them there to provide a **convenient snapshot for our users**. It is supposed to be helpful to you in case you want to know what the state of your schema was last month for example. 

Regarding your second question: I would not recommend to switch the `provider` of a datasource based on an env var. I would strive to use the same provider locally and in production. In case you really want to do this the recommended way is to use the `enabled` flag instead:
```
datasource db {
  provider = "sqlite"
  url = "file:../db/local.db"
  enabled = env("LOCAL")
}

datasource db {
  provider = "sqlite"
  url = "postgres://"
  enabled = env("PRODUCTION")
}
```

**Comment by janpio** [comment]
As a result of this we kicked of a process that might clarify what this file is for: https://github.com/prisma/specs/issues/225 Thanks for this input @Jaspaul! 🚀

**Comment by albertoperdomo** [comment]
Thanks for taking the time to report this issue!

We've recently released a Prisma Migrate Preview ([2.1.3.0 release notes](https://github.com/prisma/prisma/releases/tag/2.13.0)), which has quite a few changes compared to the previous experimental version. 

Prisma Migrate Preview does not keep a copy of the Schema file in each migration directory.

We would encourage you to try out Prisma Migrate Preview. If you come across this or any other issue in the preview version, please feel free to open a [new issue in prisma/prisma](https://github.com/prisma/prisma/issues/new/choose).
For general feedback on Prisma Migrate Preview, feel free to chime in [on this issue](https://github.com/prisma/prisma/issues/4531).
